pkg/estimator/client/cache.go

```diff
@@ -47,26 +47,13 @@ func NewSchedulerEstimatorCache() *SchedulerEstimatorCache {
 type clientWrapper struct {
 	connection *grpc.ClientConn
 	client     estimatorservice.EstimatorClient
+	closeLock  sync.Mutex
```

Comment:
why add this lock? it seems that all operations are protected by this write lock `c.lock`.

Reply:
agree, Continuing to use lock in estimatorCache is sufficient.
Would it be better to modify it like this?
```go
// EstablishConnection establishes a new gRPC connection with the specified cluster scheduler estimator.
func EstablishConnection(kubeClient kubernetes.Interface, name string, estimatorCache *SchedulerEstimatorCache, estimatorServicePrefix string, grpcConfig *grpcconnection.ClientConfig) error {
	serverAddr, err := resolveCluster(kubeClient, util.NamespaceKarmadaSystem,
		names.GenerateEstimatorServiceName(estimatorServicePrefix, name), int32(grpcConfig.TargetPort))
	if err != nil {
		return err
	}

	klog.Infof("Start dialing estimator server(%s) of cluster(%s).", serverAddr, name)
	cc, err := grpcConfig.DialWithTimeOut(serverAddr, 5*time.Second)
	if err != nil {
		klog.Errorf("Failed to dial cluster(%s): %v.", name, err)
		return err
	}
	c := estimatorservice.NewEstimatorClient(cc)
	estimatorCache.AddOrUpdateCluster(name, cc, c)
	klog.Infof("Connection with estimator server(%s) of cluster(%s) has been established.", serverAddr, name)
	return nil
}

// AddOrUpdateCluster adds or updates a grpc connection and associated client into the cache.
func (c *SchedulerEstimatorCache) AddOrUpdateCluster(name string, connection *grpc.ClientConn, client estimatorservice.EstimatorClient) {
	c.lock.Lock()
	defer c.lock.Unlock()

	if _, exist := c.estimator[name]; exist {
		_ = c.estimator[name].connection.Close()
	}
	c.estimator[name] = &clientWrapper{
		connection: connection,
		client:     client,
	}
}
```

Reply:
make sense，Thanks